docs/fsharp/style-guide/formatting.md

```diff
@@ -1438,32 +1572,100 @@ For any custom operator that starts with `*` and that has more than one characte
 
 ### Formatting record declarations
 
-For record declarations, indent `{` in type definition by four spaces, start the field list on the same line and align any members with the `{` token:
+For record declarations, by default you should indent `{` in type definition by four spaces, start the field list on the same line and align any members with the `{` token:
 
 ```fsharp
 // ✔️ OK
 type PostalAddress =
     { Address: string
       City: string
       Zip: string }
-    member x.ZipAndCity = $"{x.Zip} {x.City}"
 ```
 
-Don't place the `{` at the end of the type declaration line, and don't use `with`/`end` for members, which are redundant.
+It is also common to prefer putting brackets on their own line, with members indented by an additional four spaces:
 
 ```fsharp
-// ❌ Not OK, code formatters will reformat to the above by default
+// ✔️ OK
+type PostalAddress =
+    { 
+        Address: string
+        City: string
+        Zip: string 
+    }
+```
+
+You may also put the `{` at the end of the type declaration line like ("stroustrup" style):
+
+```fsharp
+// ✔️ OK
 type PostalAddress = {
     Address: string
     City: string
     Zip: string
-  }
+}
+```
+
+If additional members are needed, whenever possible, don't use `with`/`end`:.
+
+```fsharp
+// ✔️ OK
+type PostalAddress =
+    { Address: string
+      City: string
+      Zip: string }
+    member x.ZipAndCity = $"{x.Zip} {x.City}"
+
+// ❌ Not OK, code formatters will reformat to the above by default
+type PostalAddress =
+    { Address: string
+      City: string
+      Zip: string }
   with
     member x.ZipAndCity = $"{x.Zip} {x.City}"
   end
+  
+// ✔️ OK
+type PostalAddress =
+    { 
+        Address: string
+        City: string
+        Zip: string 
+    }
+    member x.ZipAndCity = $"{x.Zip} {x.City}"
+    
+// ❌ Not OK, code formatters will reformat to the above by default
+type PostalAddress =
+    { 
+        Address: string
+        City: string
+        Zip: string 
+    }
+    with
+        member x.ZipAndCity = $"{x.Zip} {x.City}"
+    end
+```
+
+The exception to this style rule is if you format records according to the "Stroustrup" style. In this situation, due to compiler rules, the `with` keyword is required if you want to implement an interface or add additional members:
+
+```fsharp
+// ✔️ OK
+type PostalAddress = {
+    Address: string
+    City: string
+    Zip: string
+} with
+    member x.ZipAndCity = $"{x.Zip} {x.City}"
+   
+// ❌ Not OK, this is currently invalid F# code
+type PostalAddress = {
+    Address: string
+    City: string
+    Zip: string
+} 
+member x.ZipAndCity = $"{x.Zip} {x.City}"
 ```
 
-When XML documentation is added for record fields, it becomes normal to indent and add whitespace:
+When XML documentation is added for record fields, it becomes normal to indent and add additional whitespace between members:
```

Comment:
@josh-degraw what should happen when you have a record type with XML doc and Stroustrup?
I feel like that should also go to `Aligned` of one of the fields has XML.
Thoughts?

Reply:
Personally, from a user perspective I definitely wouldn't expect it to need to change to aligned. It makes sense to not allow Cramped style there, but I think it should be seen as fine to have xml comments on the type members.

Reply:
Cramped indeed has that problem where it would be on the same line as the XML doc comment.
Stroustrup doesn't have this issue, so might be reasonable to not change it.